### DISCUSSION TITLE
thunderbird-beta and firefox-esr profiles question

### DISCUSSION THREAD

**Discussion by slowpeek** [netblue30/firejail#4492] [Q&A] [answered]
Why

```
private-opt thunderbird-beta
```

and

```
whitelist /usr/share/firefox-esr
```

firefox-beta has nothing like this.

**Comment by rusty-snake** [comment, marked as answer]
> Why
> 
> ```
> private-opt thunderbird-beta
> ```

IDK maybe an AUR package?

> and
> 
> ```
> whitelist /usr/share/firefox-esr
> ```
> 
> firefox-[*esr*] has nothing like this.

Debian? AUR?

**Reply by slowpeek** [reply]
> IDK maybe an AUR package?

I'm speaking about https://github.com/netblue30/firejail/blob/a44fa49cb6cb6a00badd25b8f56e60e701fe8cc1/etc/profile-m-z/thunderbird-beta.profile#L9

With `private-opt thunderbird-beta` the only allowed location for thunderbird-beta installation under /opt is /opt/thunderbird-beta. For example if I have it installed into /opt/mozilla/thunderbird-beta, symlink

```
/usr/bin/thunderbird-beta => /opt/mozilla/thunderbird-beta/thunderbird
/usr/local/bin/thunderbird-beta => /usr/bin/firejail
```

and run thunderbird-beta the outcome is

```
Error: no suitable /usr/bin/thunderbird-beta executable found
```

`ignore private-opt` in thunderbird-beta.local makes it work.

I just dont get it why there is such strict restriction in the stock profile. Mby it was a temporary solution to something and it should be removed by now?

**Reply by rusty-snake** [reply]
I see two options:
1. Add more to the default list (`thunderbird,mozilla`) and add a comment to use private-opt in .locals for custom dirs.
2. Unrestrict /opt

**Reply by slowpeek** [reply]
I support option 2:

- there is nothing special in thunderbird-beta compared to thunderbird for that restriction
- stock profile should not dictate installation path
